Sim/SimPapas/src/PapasAlg.cpp

```diff
@@ -42,25 +66,30 @@ StatusCode PapasAlg::execute() {
 
   // run the tools
   try {
-    for (auto tool : m_tools) {
-      // create outputs structure (eg for export tool) so that Gaudi does not fail later on
-      // at the data output stage (eg in the case where Papas does not finish simulation)
-      tool->createOutputStructures();
-    }
+    // needed so that PODIO can find structure even if run is abandoned
+    // and there are no particles created
+    m_exportTool->createOutputStructures();
+    //import to create SimParticles and links between Sim and GenParticles
+    m_importTool->run(pevent, m_importParticleLinks, m_spDetector);
+    // now do the real work
     for (auto tool : m_tools) {
       // run each tool
-      tool->run(pevent);
+      tool->run(pevent, m_spDetector);
     }
+    m_exportTool->run(pevent, m_importParticleLinks);
```

Comment:
this line sounds funny (exporting importing particles?)
what is m_importParticleLinks? 
is the name well chosen?

Reply:
No! -  name now changed.